addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -79,15 +81,69 @@ publicWidget.registry.websiteSaleCurrency = publicWidget.Widget.extend({
     },
 });
 
-function reload() {
-    if (window.location.href.match(/\?enable_editor/)) {
-        window.location.reload();
-    } else {
-        window.location.href = window.location.href.replace(/\?(enable_editor=1&)?|#.*|$/, '?enable_editor=1&');
+var shopGridOptionsCommon = {
+    /**
+     * Displays confirmation dialog  if the user tries to change product grid.
+     *
+     * @private
+     */
+    _openConfirmationDialog: function () {
+        if (!showConfirmDialog) {
+            return Promise.resolve();
+        }
+        var $content = $(qweb.render('website_sale.dialog_confirmation'));
+        return new Promise(function (resolve) {
+            new Dialog(this, {
+                title: _t('Confirmation'),
+                size: 'medium',
+                buttons: [{
+                    text: _t('Confirm'),
+                    classes: 'btn-primary',
+                    close: true,
+                    click: function () {
+                        showConfirmDialog = !$content.find('#dialogShow').is(":checked");
+                        resolve();
+                    }
+                }, {
+                    text: _t('Cancel'),
+                    close: true
+                }],
+                $content: $content,
+            }).open();
+        });
+    },
+    /**
+     * @private
+     */
+    _getShopContext: function () {
+        var $productGrid = this.$target.closest('#product_table');
+        return _.pick($productGrid.data(), function (value) {
+            return _.isNumber(value) || _.isString(value);
```

Comment:
Is not this always true?

Reply:
Table have some extra data like `snippetEditor` object.